src/app/public/modules/progress-indicator/progress-indicator.component.ts

```diff
@@ -93,15 +90,56 @@ export class SkyProgressIndicatorComponent implements OnInit, AfterContentInit,
   @Output()
   public progressChanges = new EventEmitter<SkyProgressIndicatorChange>();
 
+  public get cssClassNames(): string {
+    const classNames = [
+      `sky-progress-indicator-mode-${this.displayModeName}`
+    ];
+
+    if (this.isPassive) {
+      classNames.push('sky-progress-indicator-passive');
+    }
+
+    return classNames.join(' ');
+  }
+
+  public get displayModeName(): string {
+    if (this.displayMode === SkyProgressIndicatorDisplayMode.Vertical) {
+      return 'vertical';
+    }
+
+    return 'horizontal';
+  }
+
+  public get hasFinishButton(): boolean {
+    return this._hasFinishButton || false;
+  }
+
+  public set hasFinishButton(value: boolean) {
+    this._hasFinishButton = value;
+  }
+
+  public get itemStatuses(): SkyProgressIndicatorItemStatus[] {
+    if (!this.itemComponents) {
+      return [];
+    }
+
+    const statuses = this.itemComponents.map(c => c.status);
+
+    // Update the view whenever item statuses change.
+    this.changeDetector.markForCheck();
```

Comment:
The status marker component wasn't receiving any changes when the array of complex objects was changing. This seemed to be the only way to communicate with the HTML view (and the child status marker component) that a change had been made.

Reply:
Gotchya. I can replicate the issue. I'm still confused why its in a getter though? I think it has to do with the fact that you call this getter in a subscribe to `progressChanges` near the bottom. So it's working but maybe seems a little like a happy coincidence?

Would it fit better to move this `markForCheck()` into the `updateSteps()` method?

Reply:
Good suggestion. I'll put the change detection at the end of the `handleIncomingMessages` method, since it makes more sense to fire after an action is taken by the component.

Reply:
Love it. Thanks!